### DISCUSSION TITLE
Issue trying to play two audio files synchronously and change the current time

### DISCUSSION THREAD

**Discussion by sterudatadriven** [naudio/NAudio#1043] [Q&A]
I'm trying to make a simple C# WinForms app, based on NAudio, that allows me to play two perfectly synchronized audio files (two tracks with same duration played in unison: Basically, they should both start from the beginning at same time and end in the end together)

I also need to manipulate volume (individually, for each track) and playback position (one single control, like a button which, once clicked, makes the two tracks go forward 10 seconds, but always in unison) during playback

Everything seems to work fine, but when I programmatically change playback position, the two tracks get out of phase a little bit (with each new iteration, it gets worse)

Here's the thread opened on stackoverflow in which I highlight used code:
https://stackoverflow.com/questions/76525471/two-audio-synchronous-playback-and-change-currenttime
I also tried the solution proposed by Ryan McDonough but without any success

Thanks in advance for your attention

**Comment by markheath** [comment]
For playing audio files synchronized I would recommend mixing them in software (e.g. using the `MixingSampleProvider`) and only opening one instance of the sound card. This would allow you to control the repositioning of the source files. I usually end up making my own custom class to manage this and ensure that repositions happen outside of calls to the Read method
